src/Theme.cpp

```diff
@@ -0,0 +1,315 @@
+            g_themeDark.tab.current.close
+        },
+        // Highlighted style
+        {
+            // Background color
+            AdjustLightness2(g_themeDark.tab.current.backgroundColor, 10),
+            // Text color
+            COL_WHITE,
+            // Default close style
+            g_themeDark.tab.current.close
+        },
+        // Tab Close Circle Enabled
+        false,
+        // Tab Close Pen Width
+        1.0f,
+        // Hovered close style
+        {
+            // X color
+            COL_WHITEISH
+        },
+        // Clicked close style
+        g_themeDark.tab.hoveredClose
+    }
+};
+
+Theme g_themeDarker = {
+    // Theme Name
+    _TRN("Darker"),
+    // Main window theme
+    {
+        // Main Background Color
+        rgb_to_bgr(0x2D2D30),
+        // Main Text Color
+        COL_WHITE,
+        // Main Link Color
+        rgb_to_bgr(0x3081D4)
+    },
+    // Document style
+    {
+        // Canvas Color
+        rgb_to_bgr(0x1E1E1E),
+        // Background Color
+        g_themeDarker.mainWindow.backgroundColor,
+        // Text color
+        g_themeDarker.mainWindow.textColor
+    },
+    // Tabs
+    {
+        // Height
+        24,
+        // Current style
+        {
+            // Background color
+            rgb_to_bgr(0x007ACC),
+            // Text color
+            COL_WHITE,
+            // Default close style
+            {
+                // X color
+                rgb_to_bgr(0xD0E6F5),
+                // Circle color
+                COL_BLACK
+            }
+        },
+        // Background style
+        {
+            // Background color
+            rgb_to_bgr(0xEAEAEA),
+            // Text color
+            COL_BLACK,
+            // Default close style
+            {
+                // X color
+                COL_BLACK,
+                // Circle color
+                COL_BLACK
+            }
+        },
+        // Highlighted style
+        {
+            // Background color
+            rgb_to_bgr(0x1C97EA),
+            // Text color
+            COL_WHITE,
+            // Default close style
+            g_themeDarker.tab.current.close
+        },
+        // Tab Close Circle Enabled
+        false,
+        // Tab Close Pen Width
+        2.0f,
+        // Hovered close style
+        {
+            // X color
+            COL_WHITE,
+            // Circle color
+            COL_BLACK
+        },
+        // Clicked close style
+        g_themeDarker.tab.hoveredClose
+    }
+};
+
+// Master themes list
+Theme *g_themes[THEME_COUNT] = {
+    &g_themeLight,
+    &g_themeDark,
+    &g_themeDarker,
+};
+
+// Current theme caching
+Theme *currentTheme = NULL;
+int currentThemeIndex;
+
+Theme *GetThemeByName(char* name) {
+    for (int i = 0; i < THEME_COUNT; i++) {
+        if (str::Eq(g_themes[i]->name, name)) {
+            return g_themes[i];
+        }
+    }
+    return NULL;
+}
+
+Theme *GetThemeByIndex(int index) {
+    return g_themes[index];

```

Comment:
In places like this it sometimes makes useful to add: `CrashIf(index < 0 || index >= THEME_COUNT)`, given that C lacks array bounds checking.

CrashIf() is only compiled in debug and pre-release builds and would trigger a crash.